### PR TITLE
added route handling and added activestatus using navlink

### PR DESCRIPTION
**Pull request by arandel1**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

Marcos and I added functionality to clicking Home, List, and Manage list links in the navbar using NavLink. We added active status feature to enhance UX, so that users can see which page they are actively viewing.

## Related Issue

Issue 3: As a user, I want to be able to navigate to all the pages in the application by clicking the links in the nav bar
[#3] (https://github.com/the-collab-lab/tcl-76-smart-shopping-list/issues/3)https://github.com/orgs/the-collab-lab/projects/248?pane=issue&itemId=73841780

## Acceptance Criteria

 - [ ] The nav element is added to the Layout component
- [ ] Links to the "Home", "List", and "Add item" pages of the app are In the nav element
- [ ] The links all function as expected using the NavLink component from react-router-dom

## Type of Changes
`enhancement`

## Updates
Screen Shot 2024-08-12 at 10 13 29 AM

### Before
C08CF354-3B99-4D39-A54F-10747B2D79BD

Screen Shot 2024-08-12 at 10 14 21 AM

### CODE REVIEW

**Comment by EmmaBin** [comment]
lgtm,  I really like the enhanced UI for active and pending status.

**Review by sar-mko — APPROVED** [review]
Looks good!

**Review by EmmaBin — APPROVED** [review]

File: src/views/Layout.jsx [hunk 1]
```diff
@@ -21,20 +30,21 @@ export function Layout() {
...
 				<main className="Layout-main">
 					{user ? <SignOutButton /> : <SignInButton />}
-
 					<Outlet />
 				</main>
 				<nav className="Nav">
 					<div className="Nav-container">
-						<a href="#" className="Nav-link">
+						<NavLink to="/" style={handleActive}>
 							Home
```
Comment (mentalcaries) [inline src/views/Layout.jsx:38]: Nice implementation of this to handle the active route

**Review by mentalcaries — APPROVED** [review]
Nice work! Good clean implementation of the AC

**Comment by mentalcaries** [comment]
https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExdW1nbXV2YjA2ZmtwaTQydTRuYzRzNGFoMzJvcjUwMXlrcjhhdzFwMCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/fnK0jeA8vIh2QLq3IZ/giphy.gif

**Comment by mentalcaries** [comment]
image

**Pull request merged**